smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/PaginatorGenerator.kt

```diff
@@ -124,10 +124,10 @@ class PaginatorGenerator : KotlinIntegration {
         writer
             .dokka(
                 """
-                    $docBody
-                    @param initialRequest A [${inputSymbol.name}] to start pagination
-                    $docReturn
-                """.trimIndent()
+                    |$docBody
+                    |@param initialRequest A [${inputSymbol.name}] to start pagination
+                    |$docReturn
+                """.trimMargin()
```

Comment:
**Question**: Why is an explicit margin necessary now?

Reply:
So when you compound multiline strings like this, they don't yield the indent structure you might expect, especially if they've been `trimIndent()`ed

eg. consider the following
```kotlin
val multiline0 = """
  foo
  bar
  baz
""".trimIndent()

val multiline1 = """
  $multiline0
  qux
  qux2
""".trimIndent()
```

The value of `multiline1` is now the literal
```
  foo
bar
baz
  qux
  qux2
```
because you cut the indent on `multiline0` but then reintroduced it on the first line when you interpolate it in. The "common indent" detected by `trimIndent` is now just an empty string, because the 2nd line onward of the inserted value `multiline0` still has no indent, so the 2nd `trimIndent()` has no effect and the output is scuffed.

We were always generating a string like this before for this pagination doc, but it was hidden by the `KotlinWriter` trimming all head/tail whitespace on comments. We can't just blindly strip whitespace anymore because it's sometimes required by markdown (sublists) so some tests caught the change here.

Providing the margin lets you compose multiline strings like this more clearly/reliably.

Reply:
Oh right, I wasn't remembering that the inner strings would've already been unindented. This makes sense!